queue/abstract.lua

```diff
@@ -125,25 +107,34 @@ function tube.take(self, timeout)
     end
 end
 
+--- Checks is the task has been taken in a current session or in a session
+-- with session uuid = session_uuid.
+-- Throw an error on failure.
+local function check_task_on_taken(tube_id, task_id, session_uuid)
+    local _taken = box.space._queue_taken_2.index.task:get{tube_id, task_id}
+
+    if _taken == nil or ((session_uuid ~= nil and _taken[4] ~= session_uuid) and
```

Comment:
I suggest to put "4" to constant.

Reply:
I think this condition is overcomplicated.

consider a following way
```lua
if session_uuid == nil then
    session_uuid = session.get_uuid(connection.id())
end

if _taken == nil or  _taken[4] ~= session_uuid then
```

Reply:
> I suggest to put "4" to constant.

Yes. All of these numbers are very inconvenient when reading or writing code. I plan to replace it with some kind of `enum`, as it was done for the `status`. But this will be done by a separate PR. And these changes are beyond the scope of this task.

Reply:
> I think this condition is overcomplicated.
> 
> consider a following way
> 
> ```lua
> if session_uuid == nil then
>     session_uuid = session.get_uuid(connection.id())
> end
> 
> if _taken == nil or  _taken[4] ~= session_uuid then
> ```

Such condition allow to avoid calling `session.get_uuid()` in case of `_taken == nil`. But I think you are right and the simple condition is preferable.